### ISSUE TITLE
CLIPTextModel does not work at half precision

### ISSUE THREAD

**Issue by chinoll** [huggingface/diffusers#1214]
### Describe the bug

RuntimeError exception thrown when using CLIPTextModel.How should I use fp16?
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/xxx/anaconda3/envs/yyy/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/xxx/anaconda3/envs/yyy/lib/python3.10/site-packages/transformers/models/clip/modeling_clip.py", line 722, in forward
    return self.text_model(
  File "/home/xxx/anaconda3/envs/yyy/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/xxx/anaconda3/envs/yyy/lib/python3.10/site-packages/transformers/models/clip/modeling_clip.py", line 643, in forward
    encoder_outputs = self.encoder(
  File "/home/xxx/anaconda3/envs/yyy/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/xxx/anaconda3/envs/yyy/lib/python3.10/site-packages/transformers/models/clip/modeling_clip.py", line 574, in forward
    layer_outputs = encoder_layer(
  File "/home/xxx/anaconda3/envs/yyy/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/xxx/anaconda3/envs/yyy/lib/python3.10/site-packages/transformers/models/clip/modeling_clip.py", line 317, in forward
    hidden_states, attn_weights = self.self_attn(
  File "/home/xxx/anaconda3/envs/yyy/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/xxx/anaconda3/envs/yyy/lib/python3.10/site-packages/transformers/models/clip/modeling_clip.py", line 257, in forward
    attn_output = torch.bmm(attn_probs, value_states)
RuntimeError: expected scalar type Half but found Float
```

### Reproduction

```python
from transformers import CLIPTokenizer,CLIPTextModel
import torch
device = torch.device("cuda")
dtype = torch.float16
tokenizer = CLIPTokenizer.from_pretrained("checkpoint",subfolder="tokenizer")
text_encoder = CLIPTextModel.from_pretrained("checkpoint/text_encoder")
#text_encoder = CLIPTextModel.from_pretrained("checkpoint",subfolder="tokenizer") # throw error
text_encoder.to(device,dtype=dtype)
tokens = tokenizer("test",return_tensors="pt",padding=True)['input_ids'].to(device).long()
hidden_states = text_encoder(tokens)
```

### System Info

- `diffusers` version: 0.7.2
- Platform: Linux-5.4.0-122-generic-x86_64-with-glibc2.31
- Python version: 3.10.6
- PyTorch version (GPU?): 1.12.1 (True)
- Huggingface_hub version: 0.10.1
- Transformers version: 4.19.2
- Using GPU in script?: <fill in>
- Using distributed or parallel set-up in script?: <fill in>

**Comment by patrickvonplaten** [comment]
Hey @chinoll,

I think you have to write:

```python
text_encoder = text_encoder.half()
```

instead of `.to(...dtype=dtype)`

**Comment by chinoll** [comment]
> Hey @chinoll,
> 
> I think you have to write:
> 
> ```python
> text_encoder = text_encoder.half()
> ```
> 
> instead of `.to(...dtype=dtype)`

@patrickvonplaten not work

**Comment by patrickvonplaten** [comment]
Actually I just ran your code snippet:

```python
from transformers import CLIPTokenizer,CLIPTextModel
import torch
device = torch.device("cuda")
dtype = torch.float16
tokenizer = CLIPTokenizer.from_pretrained("CompVis/stable-diffusion-v1-4",subfolder="tokenizer")
text_encoder = CLIPTextModel.from_pretrained("CompVis/stable-diffusion-v1-4", subfolder="text_encoder")
text_encoder = CLIPTextModel.from_pretrained("checkpoint",subfolder="tokenizer") # throw error
text_encoder.to(device,dtype=dtype)
tokens = tokenizer("test",return_tensors="pt",padding=True)['input_ids'].to(device).long()
hidden_states = text_encoder(tokens)
```

and it works just fine

**Comment by lionel3** [comment]
@chinoll  Same issue with you, I fixed this by upgrading transformers from 4.19.2 to 4.24.0 .

**Comment by chinoll** [comment]
> @chinoll Same issue with you, I fixed this by upgrading transformers from 4.19.2 to 4.24.0 .

good!Problem solved
